packages/abi/src/encoder/encoder.ts

```diff
@@ -44,8 +44,8 @@ class Encoder {
     return `${inits}${closings}`;
   }
 
-  static encodeToken (token: Token, index = 0): Mediate {
-    if (!isInstanceOf(token, Token)) {
+  static encodeToken (token?: Token, index = 0): Mediate {
```

Comment:
is there a reason we're using the `token?: Token` syntax to define an optional parameter here, and we're using the `ParamType[] | undefined` syntax in the other file?

also out of curiosity, why are we setting the first parameter of `encode` and `encodeToken` as optional if they really are required?